detect_test.go

```diff
@@ -79,4 +158,25 @@ func testDetect(t *testing.T, context spec.G, it spec.S) {
 			})
 		})
 	})
+
+	context("failure cases", func() {
+		context("$COMPOSER is set to a non-existent file", func() {
+			it.Before(func() {
+				Expect(os.Chmod(workingDir, 0000)).To(Succeed())
```

Comment:
Do we expect this to happen? Would the lifecycle give us a file with this permission?